src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -2917,6 +2917,51 @@ private CorInfoTypeWithMod getArgType(CORINFO_SIG_INFO* sig, CORINFO_ARG_LIST_ST
             }
         }
 
+        private int getExactClasses(CORINFO_CLASS_STRUCT_* baseType, int maxExactClasses, CORINFO_CLASS_STRUCT_** exactClsRet)
+        {
+#if !READYTORUN
+            MetadataType type = HandleToObject(baseType) as MetadataType;
+            if (type == null)
+            {
+                return 0;
+            }
+
+            // type is already sealed, return it
+            if (_compilation.IsEffectivelySealed(type))
+            {
+                *exactClsRet = baseType;
+                return 1;
+            }
+
+            if (!type.IsInterface)
+            {
+                // TODO: handle classes
+                return 0;
+            }
+
+            TypeDesc[] implClasses = _compilation.GetImplementingClasses(type);
+            if (implClasses == null || implClasses.Length > maxExactClasses)
+            {
+                return 0;
+            }
+
+            int index = 0;
+            foreach (TypeDesc implClass in implClasses)
+            {
+                if (implClass.IsRuntimeDeterminedType)
```

Comment:
I don't think this can be true.

We should be making sure not to put canonical forms here (I assume that's what you wanted to check for). Maybe this should be `Debug.Assert(!implClass.IsCanonicalSubtype(CanonicalFormKind.Any))`.